src/controllers/transfers.js

```diff
@@ -359,126 +129,27 @@ function * processStateTransitions (tr, transfer) {
-    transfer.credits[0].account + ' : ' +
-    transfer.credits[0].amount)
-
-  // Do all static verification (signatures, validity, etc.) here
-
-  // Verify debits
-  let totalDebits = 0
-  let totalCredits = 0
-
-  transfer.debits.forEach(function (debit) {
-    if (debit.amount <= 0) {
-      throw new UnprocessableEntityError(
-        'Amount must be a positive number excluding zero.')
-    }
-    totalDebits += parseFloat(debit.amount)
-  })
-
-  transfer.credits.forEach(function (credit) {
-    if (credit.amount <= 0) {
-      throw new UnprocessableEntityError(
-        'Amount must be a positive number excluding zero.')
-    }
-    totalCredits += parseFloat(credit.amount)
-  })
-
-  if (totalCredits !== totalDebits) {
-    throw new UnprocessableEntityError(
-      'Total credits must equal total debits')
-  }
-
-  // TODO Validate that the execution_condition_fulfillment is correct
-
-  let originalTransfer, previousDebits
-  yield db.transaction(function *(transaction) {
-    originalTransfer = yield Transfer.findById(transfer.id, {transaction})
-    if (originalTransfer) {
-      log.debug('found an existing transfer with this ID')
-      previousDebits = originalTransfer.getData().debits
-
-      // This method will update the original transfer object using the new
-      // version, but only allowing specific fields to change.
-      transfer = updateTransferObject(originalTransfer, transfer)
-    } else {
-      yield validateNoDisabledAccounts(transaction, transfer)
-      // A brand-new transfer will start out as proposed
-      updateState(transfer, 'proposed')
-    }
-
-    // This method will check that any authorized:true fields added can
-    // only be added by the owner of the account
-    // _this.req.user is set by the passport middleware
-    let user = _this.req.user
-    validateAuthorizations(user && user.name, transfer, previousDebits)
-
-    yield processStateTransitions(transaction, transfer)
-
-    // Store transfer in database
-    if (originalTransfer) {
-      yield transfer.save({transaction})
-    } else {
-      yield Transfer.create(transfer, {transaction})
-    }
-
-    // Create persistent notification events. We're doing this within the same
-    // database transaction in order to maximize the reliability of the
-    // notification system. If the server crashes while trying to post a
-    // notification it should retry it when it comes back.
-    yield notificationWorker.queueNotifications(transfer, transaction)
-
-    // Start the expiry countdown if the transfer is not yet finalized
-    // If the expires_at has passed by this time we'll consider
-    // the transfer to have made it in before the deadline
-    if (!transfer.isFinalized()) {
-      yield transferExpiryMonitor.watch(transfer)
-    }
-  })
-
-  log.debug('changes written to database')
-
-  this.body = transfer.getDataExternal()
-  this.status = originalTransfer ? 200 : 201
+  transfer.id = id.toLowerCase()
```

Comment:
You only need to set this if the above `if` block does not execute. Maybe wrap this in an `else` statement, for consistency with `controllers/subscriptions.js`.